src/Xamarin.Android.Build.Tasks/Utilities/AssemblyCompression.cs

```diff
@@ -49,19 +49,25 @@ public void SetData (string sourcePath, uint descriptorIndex)
 
 		static readonly ArrayPool<byte> bytePool = ArrayPool<byte>.Shared;
 
-		public static CompressionResult Compress (AssemblyData data)
+		public static CompressionResult Compress (AssemblyData data, string outputDirectory)
 		{
 			if (data == null)
 				throw new ArgumentNullException (nameof (data));
 
+			if (String.IsNullOrEmpty (outputDirectory))
+				throw new ArgumentException ("must not be null or empty", nameof (outputDirectory));
+
+			if (!Directory.Exists (outputDirectory))
+				Directory.CreateDirectory (outputDirectory);
```

Comment:
I think you can just call `Directory.CreateDirectory (outputDirectory)`, and it doesn't throw if it exists already.